connor/main.go

```diff
@@ -0,0 +1,146 @@
+
+	ethAddr := &sonm.EthAddress{Address: crypto.PubkeyToAddress(key.PublicKey).Bytes()}
+	identityLVl := GetIdentityLvl(cfg)
+
+	dataUpdate := time.NewTicker(3 * time.Second)
+	defer dataUpdate.Stop()
+	tradeUpdate := time.NewTicker(3 * time.Second)
+	defer tradeUpdate.Stop()
+	poolTrack := time.NewTicker(900 * time.Second) //900
+	defer poolTrack.Stop()
+	poolInit := time.NewTimer(900 * time.Second)
+	defer poolInit.Stop()
+
+	snm := w.NewSNMPriceWatcher("https://api.coinmarketcap.com/v1/ticker/sonm/")
+	token := w.NewTokenPriceWatcher("https://api.coinmarketcap.com/v1/ticker/", "https://www.cryptocompare.com/api/data/coinsnapshotfullbyid/?id=")
+	reportedPool := w.NewPoolWatcher("http://178.62.225.107:3000/v1/eth/reportedhashrates/", []string{cfg.PoolAddress.EthPoolAddr})
+	avgPool := w.NewPoolWatcher("http://178.62.225.107:3000/v1/eth/avghashrateworkers/", []string{cfg.PoolAddress.EthPoolAddr + "/1"})
+	if err = snm.Update(ctx); err != nil {
+		log.Printf("cannot update snm data: %v\n", err)
+	}
+	if err = token.Update(ctx); err != nil {
+		log.Printf("cannot update token data: %v\n", err)
+	}
+	if err = reportedPool.Update(ctx); err != nil {
+		log.Printf("cannot update reportedPool data: %v\n", err)
+	}
+	if err = avgPool.Update(ctx); err != nil {
+		log.Printf("cannot update avgPool data: %v\n", err)
+	}
+
+	modules.ChargeOrdersOnce(ctx, marketClient, token, snm, balanceReply, cfg, ethAddr, identityLVl)
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Println("context done")
+			return
+
+		case <-dataUpdate.C:
+			if err = snm.Update(ctx); err != nil {
+				log.Printf(" cannot update SNM data: %v\n", err)
+			}
+			if err = token.Update(ctx); err != nil {
+				log.Printf("cannot update TOKEN data: %v\n", err)
+			}
+			go modules.CollectTokensMiningProfit(token)
+
+		case <-tradeUpdate.C:
+			modules.TradeObserve(ctx, ethAddr, dealCli, reportedPool, token, marketClient, taskCli, cfg, identityLVl)
+		case <-poolInit.C:
+			modules.SavePoolDataToDb(ctx, reportedPool, cfg.PoolAddress.EthPoolAddr)
+		case <-poolTrack.C:
+			modules.PoolTrack(ctx, reportedPool, avgPool, cfg.PoolAddress.EthPoolAddr, dealCli, marketClient)
+		}
+	}
+}
+
+func GetIdentityLvl(cfg *config.Config) sonm.IdentityLevel {
+	switch cfg.OtherParameters.IdentityForBid {
+	case 1:
+		return sonm.IdentityLevel_ANONYMOUS
+	case 2:
+		return sonm.IdentityLevel_REGISTERED
+	case 3:
+		return sonm.IdentityLevel_IDENTIFIED
+	case 4:
+		return sonm.IdentityLevel_PROFESSIONAL
+	}
+	return sonm.IdentityLevel_UNKNOWN
+}
+func newCredentials(ctx context.Context, key *ecdsa.PrivateKey) (credentials.TransportCredentials, error) {
+	_, TLSConfig, err := util.NewHitlessCertRotator(ctx, key)
+	if err != nil {
+		return nil, err
+	}
+	return util.NewTLS(TLSConfig), nil
+}
+func newNodeClientConn(ctx context.Context, creds credentials.TransportCredentials, endpoint string) (*grpc.ClientConn, error) {
```

Comment:
this function is redundant, it is a wrapper for `xgrpc.NewClient` that doesn't do, like, anything